packages/integrations/sitemap/src/index.ts

```diff
@@ -144,7 +147,11 @@ const createPlugin = (options?: SitemapOptions): AstroIntegration => {
 						return;
 					}
 
-					let urlData = generateSitemap(pageUrls, finalSiteUrl.href, opts);
+					let urlData = generateSitemap(pageUrls, finalSiteUrl.href, {
+						...opts,
+						//TODO: remove this and exclude customPages from the SitemapOptions type. Omit<> doesn't work.
+						customPages: customPagesUrls,
+					});
```

Comment:
Whether you decide it has to return something or not, it would be good to make this change because it already fixes the type error you had.

```suggestion
					let urlData = generateSitemap(pageUrls, finalSiteUrl.href, opts);
```